server-phoenix/lib/helios_web/controllers/events_api/events_api_controller.ex

```diff
@@ -0,0 +1,44 @@
+defmodule HeliosWeb.EventsAPI.EventsAPIController do
+  use HeliosWeb, :controller
+  alias Helios.{Repo, Event}
+
+  def handle(conn, params) do
+    cond do
+      params["version"] == "1" -> api_v1(conn, params)
+      true -> send_resp(conn, 404, "Not Found")
+    end
+  end
+
+  defp api_v1(conn, params) do
+    case params["event_type"] do
+      "github_commit" -> post_commit(conn, params)
+      _ -> send_resp(conn, 400, "Bad Request")
+    end
+  end
+
+  defp post_commit(conn, params) do
+    unless(
+      Event
+      |> Event.commits()
+      |> Event.with_external_id(params["event_id"])
+      |> Repo.exists?()
+    ) do
+      Repo.insert!(%Event{
+        source: :github_commit,
+        external_id: params["event_id"],
+        source_author: params["author"]
+      })
+      |> publish
+    end
+
+    send_resp(conn, 201, "Created")
```

Comment:
I wonder if there is value for an HTTP result if it already exists. I feel like that could be useful for debugging and probably isn't an error/failure since its a common case when pushing up a branch. 409 Conflict seems out. 303 See Other? We don't really want to redirect though. :thinking:

Reply:
208 Already Reported (WebDAV; RFC 5842) sounds like it makes sense

Reply:
Yup 208 looks best suited. Added.